src/Interpreters/Aggregator.cpp

```diff
@@ -237,8 +469,7 @@ class CompiledAggregateFunctionsHolder final : public CompiledExpressionCacheEnt
 
 #endif
 
-Aggregator::Aggregator(const Params & params_)
-    : params(params_)
+Aggregator::Aggregator(const Params & params_) : params(params_), stats_updater(std::make_unique<StatisticsUpdater>(*this))
```

Comment:
Passing `this` from ctor looks dangerous because object is not fully constructed yet, but seems in this case it's fine, we will just store refernce to this somewhere

https://isocpp.org/wiki/faq/ctors#using-this-in-ctors

Reply:
Also, maybe not to construct `stats_updater` at all if statistics in disabled, just leave `nullptr`?

Reply:
> Also, maybe not to construct stats_updater at all if statistics in disabled, just leave nullptr?

done